src/ucp/api/ucp.h

```diff
@@ -1282,11 +1294,22 @@ void ucp_request_release(void *request);
  * @param [in]  worker       UCP worker.
  * @param [in]  request      Non-blocking request to cancel.
  *
- *  A request can either be completed or canceled, but not both. After calling
- * this function, the request will complete regardless of what the remote side
- * is doing. If the request is completed successfully, the completion callback
- * will be called with the status parameter equals to UCS_OK, and in case it's
- * canceled the status would be UCS_ERR_CANCELED.
+ * This routine tries to cancels an outstanding communication request.  After
+ * calling this routine, the @a request will be in completed or canceled (but
+ * not both) state regardless of the status of the target endpoint associated
+ * with the communication request.  If the request is completed successfully,
+ * the @ref ucp_send_callback_t "send" or @ref ucp_tag_recv_callback_t
+ * "receive" completion callbacks (based on the type of the request) will be
+ * called with the @a status argument of the callback set to UCS_OK, and in a
+ * case it is canceled the @a status argument is set to UCS_ERR_CANCELED.  It is
+ * important to note that in order to release the request back to the library
+ * the application is responsible to call @ref ucp_request_release
+ * "ucp_request_release()".
+ *
+ * @note This routine does not undo any memory alternations associated with the
```

Comment:
a bit confusing - a memory operation can either be completed or canceled. Why would it be meaningful to "undo" it?